crates/context_aware_config/src/api/dimension/handlers.rs

```diff
@@ -131,3 +144,54 @@ async fn get(
 
     Ok(HttpResponse::Ok().json(dimensions_with_mandatory))
 }
+
+pub fn delete_dimension_helper(
+    name: String,
+    user: User,
+    conn: &mut PooledConnection<ConnectionManager<PgConnection>>,
+) -> superposition::Result<()> {
+    use dimensions::dsl;
+    diesel::update(dsl::dimensions)
+        .filter(dsl::dimension.eq(&name))
+        .set((
+            dsl::last_modified_at.eq(Utc::now().naive_utc()),
+            dsl::last_modified_by.eq(user.get_email()),
+        ))
+        .execute(conn)?;
+    let deleted_row =
+        delete(dsl::dimensions.filter(dsl::dimension.eq(&name))).execute(conn);
+    match deleted_row {
+        Ok(0) => Err(not_found!("Dimension `{}` doesn't exists", name)),
+        Ok(_) => {
+            log::info!("{name} dimension deleted by {}", user.get_email());
+            Ok(())
+        }
+        Err(e) => {
+            log::error!("dimension delete query failed with error: {e}");
+            Err(unexpected_error!("Something went wrong."))
+        }
+    }
+}
+
+#[delete("/{name}")]
+async fn delete_dimension(
+    path: Path<DeleteReq>,
+    user: User,
+    mut db_conn: DbConnection,
+) -> superposition::Result<HttpResponse> {
+    let name = path.into_inner().into();
+    fetch_dimension(&name, &mut db_conn)?;
```

Comment:
lets return 204 when its Not Found over here, for other errors, we can still return back the error from here

Reply:
Why 204? Not found is fine i guess here?

Reply:
@ayushjain17 its debatable , either way is fine 404 or 204
https://stackoverflow.com/questions/6439416/status-code-when-deleting-a-resource-using-http-delete-for-the-second-time

Reply:
@mahatoankitkumar 204 is No Content, which is the standard response on a successful delete,
so when someone is trying to delete an object which does not exist in the first place, services generally still return 204 (saying that the delete operation is done successfully) instead of throwing an error

though, we need to still return error response in case its not a Not Found error

Reply:
@ayushjain17 As @pratikmishra356 said its debatable. There are more documentation in support of 404 in such cases
https://medium.com/@satokenta940/the-complete-guide-to-understanding-http-delete-requests-04a041822784#:~:text=404%20Not%20Found%3A%20The%20resource,or%20has%20already%20been%20deleted.
https://cloud.google.com/storage/docs/xml-api/delete-object#:~:text=If%20you%20make%20a%20DELETE,NoSuchKey%20in%20the%20Code%20element.

Reply:
https://http.dev/delete

Reply:
https://www.mscharhag.com/api-design/rest-deleting-resources

Reply:
https://dev.to/cesareferrari/deleting-a-resource-via-http-delete-5hd9

Reply:
We should stick with 404 when the key is not found. We will keep this consistent across all apis.